### PR TITLE
Update Unix build docs removing mono requirement

### PR DESCRIPTION
**Pull request by jasonwilliams200OK**
Mono and PCL reference assemblies are not required anymore to build
CoreFX on Unix - #6077.

Also, PDBs are now generated by Roslyn for Unix, so removed that from
known issues - dotnet/roslyn#2449.

cc @mellinoe, @stephentoub

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @ellismg

File: Documentation/building/unix-instructions.md [hunk 1]
```diff
@@ -1,11 +1,9 @@
 Building CoreFX on FreeBSD, Linux and OS X
 ==========================================
 
-CoreFx can be built on top of current [Mono CI builds](#installing-mono-packages) or a direct [build/install of Mono](http://www.mono-project.com/docs/compiling-mono/). It builds using MSBuild and Roslyn and requires changes that have not yet made it to official released builds.
+Building CoreFX is pretty striaght-forward. Clone the repo and run the build script.
```
Comment (justinvp) [inline Documentation/building/unix-instructions.md]: Nit: straightforward

Comment (jasonwilliams200OK) [inline Documentation/building/unix-instructions.md]: Fixed thanks (although some do prefer straight-forward for uncomplicated stuff :smile:)

**Comment by stephentoub** [comment]
@mmitche, another one of those "failed to mkdirs" errors: http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/ubuntu_debug_prtest/2939/

**Comment by mmitche** [comment]
@stephentoub Thanks.  Machine got deleted.

**Comment by mellinoe** [comment]
> Building CoreFX is pretty straightforward. Clone the repo and run the build script.

:+1: 

Do we mention the installation pre-reqs for the build anywhere? I know we require cmake, for example, but I don't know if there's any others. I guess most distros come with these pre-reqs installed already so it's not worth mentioning?

**Comment by jasonwilliams200OK** [comment]
> Do we mention the installation pre-reqs for the build anywhere?

That would be nice!

> I guess most distros come with these pre-reqs installed already so it's not worth mentioning?

For instance, most of the Linux distros don't have LLVM installed OOTB (as they prefer GNU GCC).

I think if we treat build CoreCLR from source as a prerequisite of building CoreFX, then those will supersede.

Otherwise, to build CoreFX, we do need:

* curl (devel)
* icu
* llvm
* lldb

Also, we can explain the command line options in this doc such as `./build.sh native`, `debug`, `release` etc.

**Comment by jasonwilliams200OK** [comment]
@mellinoe, with 940bb7, the doc looks like: https://github.com/dotnet/corefx/blob/940bbc7/Documentation/building/unix-instructions.md.

**Comment by mellinoe** [comment]
The latest commit looks great to me.

**Pull request merged**